evm.md

```diff
@@ -1863,15 +1870,27 @@ Overall Gas
     rule <k> #memory [ OP ] => #memory(OP, MU) ~> #deductMemory ... </k>
          <memoryUsed> MU </memoryUsed>
 
-    syntax InternalOp ::= "#gas"    "[" OpCode "]" | "#deductGas" | "#deductMemoryGas"
+    syntax InternalOp ::= "#gas"    "[" OpCode "]" | "#deductGas" | "#deductMemoryGas" | "#resolveGas"
                         | "#memory" "[" OpCode "]" | "#deductMemory"
  // ----------------------------------------------------------------
     rule <k> MU':Int ~> #deductMemory => (Cmem(SCHED, MU') -Int Cmem(SCHED, MU)) ~> #deductMemoryGas ... </k>
          <memoryUsed> MU => MU' </memoryUsed> <schedule> SCHED </schedule>
 
-    rule <k> G:Int ~> (#deductMemoryGas => #deductGas)   ... </k> //Required for verification
-    rule <k> G:Int ~> #deductGas => #end EVMC_OUT_OF_GAS ... </k> <gas> GAVAIL                  </gas> requires GAVAIL <Int G
-    rule <k> G:Int ~> #deductGas => .                    ... </k> <gas> GAVAIL => GAVAIL -Int G </gas> requires GAVAIL >=Int G
+    rule <k> G:Int ~> #deductMemoryGas => .  ... </k>
+         <gas> #gas(_, _, MEMGAS => MEMGAS +Int G) </gas>
+    rule <k> G:Int ~> #deductGas => . ... </k>
+         <gas> #gas(_, EXECGAS => EXECGAS +Int G, _) </gas>
+
+    rule <k> #resolveGas => . ... </k>
+         <gas> #gas(G:Int , EXECGAS, MEMGAS) => #gas(G -Int EXECGAS -Int MEMGAS, 0, 0) </gas>
+       requires G >=Int (EXECGAS +Int MEMGAS)
+
+    rule <k> #resolveGas => #end EVMC_OUT_OF_GAS ... </k>
+         <gas> #gas(G:Int , EXECGAS, MEMGAS) </gas>
+       requires notBool (G >=Int (EXECGAS +Int MEMGAS))
+
+    rule <k> #resolveGas => . ... </k>  // For Verification
+         <gas> #gas(InfGas, EXECGAS, MEMGAS) </gas>
```

Comment:
No need to say that this is for Verification, it will also be for other things (eg. when we want to compute concrete gas usage).